CMakeLists.txt

```diff
@@ -1533,11 +1531,19 @@ if(ENGINEPRIME)
 
     # Since we have built libdjinterop from sources as a static library, its
     # transitive dependencies are not automatically recognised.  libdjinterop
-    # depends on zlib and sqlite3.  Mixxx already has a dependency on sqlite3,
-    # but not zlib, so we explicitly add that here.
+    # depends on zlib and optionally sqlite3.  If libdjinterop was configured
+    # to depend on system SQLite, Mixxx will already have the dependency.
+    # But it does not have zlib, so we explicitly add that here.
     find_package(ZLIB 1.2.8 REQUIRED)
     target_link_libraries(mixxx-lib PUBLIC ${ZLIB_LIBRARIES})
   endif()
+
+  # Include conditional sources only required with Engine Prime export support.
+  target_sources(mixxx-lib PRIVATE
+    src/library/export/dlglibraryexport.cpp
+    src/library/export/engineprimeexportjob.cpp
+    src/library/export/libraryexporter.cpp)
+  target_compile_definitions(mixxx-lib PUBLIC __ENGINEPRIME__)
```

Comment:
Can we rename the compile directive to __DJINTEROP__ or something like this. Since the library is supposed to become generic using engineprime here will only be confusing.

Reply:
No objections to this, although I'll probably do it via a separate PR to minimise the changes here.

Reply:
I disagree. We can rename that flag when libdjinterop supports other formats *and* we use it for this purpose in Mixxx. Until then, `__ENGINEPRIME__` makes clear what this flag affects and that disabling is has no influence on the Rekordbox, Reaktor or Serato integration.

Reply:
Whichever way the discussion goes, could we agree that any hypothetical renaming of the compile-time directive could happen separately from this PR?

I don't have a strong opinion on what it should be called right now, but I would like to keep the number of blocking issues for this PR low.